modules/presets/secure-node.nix

```diff
@@ -65,6 +65,45 @@ in {
       discover = false;
       addresstype = "bech32";
       dbCache = 1000;
+      rpc.users.privileged = {
+        name = "bitcoinrpc";
+        # Placeholder to be sed'd out by bitcoind preStart
+        passwordHMAC = "bitcoin-HMAC-privileged";
+      };
+      rpc.users.public = {
+        name = "publicrpc";
+        # Placeholder to be sed'd out by bitcoind preStart
+        passwordHMAC = "bitcoin-HMAC-public";
+        rpcwhitelist = [
+          "echo"
+          "getinfo"
+          "getblock"
+          "getblockchaininfo"
+          "getbestblockhash"
+          "getblockcount"
+          "getblockhash"
+          "getblockheader"
+          "getchaintips"
+          "getdifficulty"
+          "getnetworkinfo"
+          "getmempoolinfo"
+          "getmempoolentry"
+          "getrawmempool"
+          "gettxout"
+          "gettxoutproof"
+          "gettxoutsetinfo"
+          "verifytxoutproof"
+          "createrawtransaction"
+          "decoderawtransaction"
+          "decodescript"
+          "getrawtransaction"
+          "sendrawtransaction"
+          "estimatefee"
+          "estimatepriority"
+          "estimatesmartfee"
+          "estimatesmartpriority"
```

Comment:
Few more harmless calls to consider:

* `validateaddress`
* `testmempoolaccept`
* `signrawtransactionwithkey` - the key is provided in the call, so stealing is not possible
* `scantxoutset` - I assume this operation takes some time, so maybe it could be used for DoS, but probably not too huge issue?

Reply:
Incorporated these harmless calls in 88c7348105bdbf0a1e7947ddb32c0a3263b674f7. Thank you for the suggestions.

Reply:
estimatepriority and estimatesmartpriority were removed in [0.15](https://bitcoincore.org/en/releases/0.15.0/).